### PR TITLE
ADD documentation/Tutorial for learn then test

### PR DESCRIPTION
**Pull request by SZiane**
# Description
I want to add documentation and a tutorial for learn then test to be clear on what does learn then test. To do that i propose to add some documentation on the mapie homepage and a notebook to explain the ltt procedure.

## Type of change
- This change requires a documentation update

# Checklist

- [x] I have read the [contributing guidelines](https://github.com/simai-ml/MAPIE/blob/master/CONTRIBUTING.rst)
- [x] I have updated the [HISTORY.rst](https://github.com/simai-ml/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/simai-ml/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `make lint`
- [x] Typing passes successfully : `make type-check`
- [x] Unit tests pass successfully : `make tests`
- [x] Coverage is 100% : `make coverage`
- [x] Documentation builds successfully : `make doc`

### CODE REVIEW

File: doc/theoretical_description_multilabel_classification.rst [hunk 1]
```diff
@@ -31,7 +31,8 @@ on the recall. RCPS, LTT and CRC give three slightly different guarantees:
 - LTT:
 
 .. math::
-    \mathbb{P}(R(\mathcal{T}_{\lambda_{\lambda\in\hat{\Lambda}}}) \leq \alpha ) \geq 1 - \delta
+    \mathbb{P}(R(\mathcal{T}_\lambda) \leq \alpha ) \geq 1 - \delta & p_\lambda \leq \frac{\delta}{\lvert \Lambda \rvert}
+
```
Comment (thibaultcordier) [inline doc/theoretical_description_multilabel_classification.rst:35]:
```suggestion
    \mathbb{P}(R(\mathcal{T}_{\hat{\lambda}}) \leq \alpha ) \geq 1 - \delta \quad \texttt{with} \quad p_\lambda \leq \frac{\delta}{\lvert \Lambda \rvert}
```

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py [hunk 2]
```diff
@@ -208,3 +211,62 @@
         fontsize=20
     )
 plt.show()
+
+##############################################################################
+# 4. Learn Then Test
+# ------------------
+# In this part, we will use learn then test to control precision.
```
Comment (thibaultcordier) [inline examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py:218]:
```suggestion
# In this part, we will use LTT to control precision.
```

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py [hunk 3]
```diff
@@ -208,3 +211,62 @@
...
+# each $\lambda$ should satisfy FWER control. :math:`\alpha`
+# is the desired risk.
+
+# Notice that the procedure will diligently examine each :math: `\lambda`
+# such that the risk remains below level :math: `\alpha`, meaning not
+# every : math:`\lambda` will be considered.
+# This means that a for a :math:`\lambda` such that risk is below
+# :math:`\alpha`
+# doesn't necessarly pass the FWER control! This is what we are going to
+# explore.
```
Comment (thibaultcordier) [inline examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py:236]:
```suggestion
# The procedure consist in multiple hypothesis testing. This is why the output
# of this procedure isn't reduce to one value of :math:`\lambda`.
#
# More precisely, we look after all the :math:`\lambda` that sastisfy the
# following:
# :math:`\mathbb{P}(R(\mathcal{T}_{\lambda}) \leq \alpha ) \geq 1 - \delta`,
# where :math:`R(\mathcal{T}_{\lambda})` is the risk we want to control and
# each :math:`\lambda` should satisfy FWER control. :math:`\alpha`
# is the desired risk.
#
# Notice that the procedure will diligently examine each :math:`\lambda`
# such that the risk remains below level :math:`\alpha`, meaning not
# every :math:`\lambda` will be considered.
# This means that a for a :math:`\lambda` such that risk is below
# :math:`\alpha`
# doesn't necessarly pass the FWER control! This is what we are going to
# explore.
```

Comment (thibaultcordier) [reply examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py:236]: @SZiane, Can you check that these typos are also present in the notebook? If so, can you correct them?

**Review by thibaultcordier — COMMENTED** [review]
Thank you for these changes. Here my suggestions:
- Add you changes in the HISTORY.rst
- Is it possible to create two notebooks, one for CRC and RCPS, and the other for LTT? In the notebook collection, users can see both, and it might be clearer to understand that there are two families of methods.
- In the same way, is it possible to create two examples, one for CRC and RCPS, and the other for LTT? And present them in two different tutorials in the documentation section of multilabel classification (one for recall control, the other for precision control)?

File: doc/theoretical_description_multilabel_classification.rst (hunk 1 above)
Comment (SZiane) [reply doc/theoretical_description_multilabel_classification.rst]: Done

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py (hunk 2 above)
Comment (SZiane) [reply examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py]: Done

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py (hunk 3 above)
Comment (SZiane) [reply examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py]: Done

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py [hunk 4]
```diff
@@ -90,7 +93,11 @@
 plt.show()
 
 ##############################################################################
-# 2. Fitting MapieMultiLabelClassifier
+# 2 Recall control risk with CRC and RCPS
+# ---------------------------------------
+##############################################################################
+
+# 2.1 Fitting MapieMultiLabelClassifier
 # ------------------------------------
```
Comment (thibaultcordier) [inline examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py:101]:
```suggestion
# 2 Recall control risk with CRC and RCPS
# ---------------------------------------
# 2.1 Fitting MapieMultiLabelClassifier
# -------------------------------------
```

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py [hunk 5]
```diff
@@ -208,3 +215,74 @@
         fontsize=20
     )
 plt.show()
+
+##############################################################################
+# 3. Precision control risk with LTT
+# ------------------
+##############################################################################
+# 3.1 Fitting MapieMultilabelClassifier
+# -------------------------------------
```
Comment (thibaultcordier) [inline examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py:224]:
```suggestion
##############################################################################
# 3. Precision control risk with LTT
# ----------------------------------
# 3.1 Fitting MapieMultilabelClassifier
# -------------------------------------
```

**Review by thibaultcordier — COMMENTED** [review]
I have a few suggestions for improving the documentation.

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py [hunk 6]
```diff
@@ -208,3 +215,74 @@
...
+#
+# In this part, we will use LTT precision.
```
Comment (thibaultcordier) [inline examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py:226]:
```suggestion
# In this part, we will use LTT to control precision.
```

File: doc/theoretical_description_multilabel_classification.rst [hunk 7]
```diff
@@ -175,14 +185,17 @@ hypothesis testing. We can express the goal of the procedure as follows:
...
-\{(x_1, y_1), \dots, (x_n, y_n)\}`.
+:math:`\{(x_1, y_1), \dots, (x_n, y_n)\}`.
+
 2: For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
-:math:`\mathbb{H}_j: R(\lambda_j)>\alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
+:math:`\mathbb{H}_j: R(\lambda_j) > \alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
 is controlled.
-3: For each null hypothesis, we compute a valid p-value using a concentration inequality. Here we choose to compute the Hoeffding-Bentkus p-value
+
+3: For each null hypothesis, we compute a valid p-value using a concentration inequality :math: `p_\lambda_j`. Here we choose to compute the Hoeffding-Bentkus p-value
```
Comment (thibaultcordier) [inline doc/theoretical_description_multilabel_classification.rst:194]:
```suggestion
3: For each null hypothesis, we compute a valid p-value using a concentration inequality :math:`p_\lambda_j`. Here we choose to compute the Hoeffding-Bentkus p-value
```

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py [hunk 8]
```diff
@@ -208,3 +213,73 @@
...
+# of this procedure isn't reduce to one value of :math:`\lambda`.
+#
+# More precisely, we look after all the :math:`\lambda` that sastisfy the
+# following:
+# :math:`\mathbb{P}(R(\mathcal{T}_{\lambda}) \leq \alpha ) \geq 1 - \delta`,
+# where :math:`R(\mathcal{T}_{\lambda})` is the risk we want to control and
+# each :math:`\lambda`` should satisfy FWER control.
+# :math:`\alpha` is the desired risk.
+#
+# Notice that the procedure will diligently examine each :math: `\lambda`
```
Comment (thibaultcordier) [inline examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py:235]:
```suggestion
# Notice that the procedure will diligently examine each :math:`\lambda`
```

File: notebooks/classification/tutorial_multilabel_classification_precision.ipynb [hunk 9]
```diff
@@ -0,0 +1,260 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this tutorial, we use LTT to create prediction sets such that\n",
+    "precision score is controlled. The LTT method isimplemented in MAPIE,\n",
```
Comment (thibaultcordier) [inline notebooks/classification/tutorial_multilabel_classification_precision.ipynb:9]:
```suggestion
    "precision score is controlled. The LTT method is implemented in MAPIE,\n",
```

Comment (thibaultcordier) [reply notebooks/classification/tutorial_multilabel_classification_precision.ipynb:9]: @SZiane, can you check all the cells in the notebook to make sure there are no typos?

**Review by thibaultcordier — CHANGES_REQUESTED** [review]
Good work on this PR. It reveals only a few typos.

File: doc/theoretical_description_multilabel_classification.rst (hunk 7 above)
Comment (SZiane) [reply doc/theoretical_description_multilabel_classification.rst:194]: Done

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py (hunk 8 above)
Comment (SZiane) [reply examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py:235]: done

File: notebooks/classification/tutorial_multilabel_classification_precision.ipynb (hunk 9 above)
Comment (SZiane) [reply notebooks/classification/tutorial_multilabel_classification_precision.ipynb:9]: Done

File: doc/theoretical_description_multilabel_classification.rst [hunk 10]
```diff
@@ -175,14 +185,17 @@ hypothesis testing. We can express the goal of the procedure as follows:
...
-\{(x_1, y_1), \dots, (x_n, y_n)\}`.
+:math:`\{(x_1, y_1), \dots, (x_n, y_n)\}`.
+
 2: For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
-:math:`\mathbb{H}_j: R(\lambda_j)>\alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
+:math:`\mathbb{H}_j: R(\lambda_j) > \alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
 is controlled.
-3: For each null hypothesis, we compute a valid p-value using a concentration inequality. Here we choose to compute the Hoeffding-Bentkus p-value
+
+3: For each null hypothesis, we compute a valid p-value using a concentration inequality :math:`p_\lambda_j`. Here we choose to compute the Hoeffding-Bentkus p-value
```
Comment (LacombeLouis) [inline doc/theoretical_description_multilabel_classification.rst:194]: There is an error in the run of this code (check build of the documentation).

File: doc/theoretical_description_multilabel_classification.rst [hunk 11]
```diff
@@ -175,14 +185,17 @@ hypothesis testing. We can express the goal of the procedure as follows:
 In order to find all the parameters :math:`\lambda` that satisfy the above condition, Learn Then Test propose to do the following:
```
Comment (LacombeLouis) [inline doc/theoretical_description_multilabel_classification.rst:185]: In order to find all the parameters :math:`\lambda` that satisfy the above condition, the Learn Then Test framework proposes to do the following:

File: doc/theoretical_description_multilabel_classification.rst [hunk 12]
```diff
@@ -175,14 +185,17 @@ hypothesis testing. We can express the goal of the procedure as follows:
...
 
 1: First across the collections of functions :math:`(T_\lambda)_{\lambda\in\Lambda}`, we estimate the risk on the calibration data
```
Comment (LacombeLouis) [inline doc/theoretical_description_multilabel_classification.rst:187]: These are not bullet points, not sure if that's the aim, but I don't find it very easy to read. Maybe put the numbers in bold?

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py [hunk 13]
```diff
@@ -208,3 +213,73 @@
...
+
+r_hat = mapie_clf.r_hat
+idx_max = np.argmin(r_hat[valid_index])
+
+##############################################################################
+# 3.2 Results
+# -----------
+# We can see that not all :math:`\lambda` such that risk is below the orange
+# line are choosen by the procedure. Otherwise, all the lambdas that are
+# in the red rectangle verify fwer control and allow to control precision
```
Comment (LacombeLouis) [inline examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py:272]: *fewer

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py [hunk 14]
```diff
@@ -208,3 +213,73 @@
         fontsize=20
     )
 plt.show()
```
Comment (LacombeLouis) [inline examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py:215]: This graph is too small. Difficult to see.

**Review by LacombeLouis — CHANGES_REQUESTED** [review]
Good addition to the documentation! LTT is going to be 🔥

I don't get why we have notebook and examples. To me it seems like it's the same thing? Between the [notebooks/classification/tutorial_multilabel_classification_recall.ipynb](https://github.com/scikit-learn-contrib/MAPIE/pull/347/files#diff-dc6dc486cd38bf782797ecf82e57c24cf6a2207bbd1d971c3bb110d9ca4bf46b) + [notebooks/classification/tutorial_multilabel_classification_precision.ipynb](https://github.com/scikit-learn-contrib/MAPIE/pull/347/files#diff-d0b34d97fb8a49e1c8a332da4549523eb59a1dbfa072dbb7b93cb6a6d73b2d93) = examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py

No?

File: doc/theoretical_description_multilabel_classification.rst (hunk 10 above)
Comment (thibaultcordier) [reply doc/theoretical_description_multilabel_classification.rst:194]:
```suggestion
3: For each null hypothesis, we compute a valid p-value using a concentration inequality :math:`p_{\lambda_j}`. Here we choose to compute the Hoeffding-Bentkus p-value
```

File: doc/theoretical_description_multilabel_classification.rst (hunk 11 above)
Comment (thibaultcordier) [reply doc/theoretical_description_multilabel_classification.rst:185]:
```suggestion
In order to find all the parameters :math:`\lambda` that satisfy the above condition, the Learn Then Test framework proposes to do the following:
```

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py (hunk 13 above)
Comment (thibaultcordier) [reply examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py:272]:
```suggestion
# in the red rectangle verify fewer control and allow to control precision
```

File: notebooks/classification/tutorial_multilabel_classification_precision.ipynb [hunk 15]
```diff
@@ -0,0 +1,260 @@
+{
```
[the remark this answers is not in the document]
Comment (thibaultcordier) [reply notebooks/classification/tutorial_multilabel_classification_precision.ipynb:1]: I have proposed to @SZiane the following:
- add two notebooks, one for precision, the other for recall
- propose one exemple in the documentation directly (via Tutorial for multilabel-classification)

By doing that, users can run the notebook for their purpose (control recall or precision). But we continue to propose a comparision directly in the tutorial/documentation to explain how MAPIE can help them.

File: doc/theoretical_description_multilabel_classification.rst [hunk 16]
```diff
@@ -166,6 +167,15 @@ With :
 
 3. Learn Then Test
 ------------------
+
+3.1. General settings
```
[the remark this answers is not in the document]
Comment (SZiane) [reply doc/theoretical_description_multilabel_classification.rst:171]: References are at the end of the document!

File: doc/theoretical_description_multilabel_classification.rst (hunk 10 above)
Comment (SZiane) [reply doc/theoretical_description_multilabel_classification.rst]: Done

File: doc/theoretical_description_multilabel_classification.rst (hunk 11 above)
Comment (SZiane) [reply doc/theoretical_description_multilabel_classification.rst]: done

File: doc/theoretical_description_multilabel_classification.rst (hunk 12 above)
Comment (SZiane) [reply doc/theoretical_description_multilabel_classification.rst]: Done, adding bullet points

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py (hunk 13 above)
Comment (SZiane) [reply examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py]: It is actually fwer for family wise error rate. I will change it to family wise error rate to be more precise

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py [hunk 17]
```diff
@@ -208,3 +213,73 @@
...
+lambdas = mapie_clf.lambdas[valid_index]
+
+mini = lambdas[np.argmin(lambdas)]
+maxi = lambdas[np.argmax(lambdas)]
+
+r_hat = mapie_clf.r_hat
+idx_max = np.argmin(r_hat[valid_index])
+
+##############################################################################
+# 3.2 Results
```
[the remark this answers is not in the document]
Comment (SZiane) [reply examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py]: Tell me if title of new part is satisfying.

File: examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py (hunk 14 above)
Comment (SZiane) [reply examples/multilabel_classification/1-quickstart/plot_tutorial_multilabel_classification.py:215]: Done.

File: doc/theoretical_description_multilabel_classification.rst (hunk 16 above)
[the remark this answers is not in the document]
Comment (thibaultcordier) [reply doc/theoretical_description_multilabel_classification.rst:171]: The suggestion was to introduce this part with the reference of the LLT method :)

File: doc/theoretical_description_multilabel_classification.rst [hunk 18]
```diff
@@ -166,23 +167,35 @@ With :
...
 hypothesis testing. We can express the goal of the procedure as follows:
 
 .. math::
     \mathbb{P}(R(\mathcal{T}_{\lambda}) \leq \alpha ) \geq 1 - \delta
 
-In order to find all the parameters :math:`\lambda` that satisfy the above condition, Learn Then Test propose to do the following:
+In order to find all the parameters :math:`\lambda` that satisfy the above condition, the Learn Then Test framework proposes to do the following:
+
+- First across the collections of functions :math:`(T_\lambda)_{\lambda\in\Lambda}`, we estimate the risk on the calibration data
+:math:`\{(x_1, y_1), \dots, (x_n, y_n)\}`.
```
Comment (thibaultcordier) [inline doc/theoretical_description_multilabel_classification.rst:188]:
```suggestion
- First across the collections of functions :math:`(T_\lambda)_{\lambda\in\Lambda}`, we estimate the risk on the calibration data
  :math:`\{(x_1, y_1), \dots, (x_n, y_n)\}`.
```

File: doc/theoretical_description_multilabel_classification.rst [hunk 19]
```diff
@@ -166,23 +167,35 @@ With :
...
 
-1: First across the collections of functions :math:`(T_\lambda)_{\lambda\in\Lambda}`, we estimate the risk on the calibration data
-\{(x_1, y_1), \dots, (x_n, y_n)\}`.
-2: For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
-:math:`\mathbb{H}_j: R(\lambda_j)>\alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
+- For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
+:math:`\mathbb{H}_j: R(\lambda_j) > \alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk
```
Comment (thibaultcordier) [inline doc/theoretical_description_multilabel_classification.rst:191]:
```suggestion
- For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
  :math:`\mathbb{H}_j: R(\lambda_j) > \alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
```

File: doc/theoretical_description_multilabel_classification.rst [hunk 20]
```diff
@@ -166,23 +167,35 @@ With :
...
-\{(x_1, y_1), \dots, (x_n, y_n)\}`.
-2: For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
-:math:`\mathbb{H}_j: R(\lambda_j)>\alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
+- For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
+:math:`\mathbb{H}_j: R(\lambda_j) > \alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
 is controlled.
-3: For each null hypothesis, we compute a valid p-value using a concentration inequality. Here we choose to compute the Hoeffding-Bentkus p-value
+
+- For each null hypothesis, we compute a valid p-value using a concentration inequality :math:`p_{\lambda_j}`. Here we choose to compute the Hoeffding-Bentkus p-value
 introduced in the paper [3].
```
Comment (thibaultcordier) [inline doc/theoretical_description_multilabel_classification.rst:195]:
```suggestion
- For each null hypothesis, we compute a valid p-value using a concentration inequality :math:`p_{\lambda_j}`. Here we choose to compute the Hoeffding-Bentkus p-value
  introduced in the paper [3].
```

File: doc/theoretical_description_multilabel_classification.rst [hunk 21]
```diff
@@ -166,23 +167,35 @@ With :
...
-4: Return :math:`\hat{\Lambda} =  \mathbb{A}(\{p_j\}_{j\in\{1,\dots,lvert \Lambda \rvert})`, where :math:`\mathbb{A}`, is an algorithm
-that controls the family-wise-error-rate (FWER).
+
+- Return :math:`\hat{\Lambda} =  \mathcal{A}(\{p_j\}_{j\in\{1,\dots,\lvert \Lambda \rvert})`, where :math:`\mathcal{A}`, is an algorithm
+that controls the family-wise-error-rate (FWER), for example bonferonni correction.
```
Comment (thibaultcordier) [inline doc/theoretical_description_multilabel_classification.rst:198]:
```suggestion
- Return :math:`\hat{\Lambda} =  \mathcal{A}(\{p_j\}_{j\in\{1,\dots,\lvert \Lambda \rvert})`, where :math:`\mathcal{A}`, is an algorithm
  that controls the family-wise-error-rate (FWER), for example bonferonni correction.
```

File: doc/theoretical_description_multilabel_classification.rst (hunk 12 above)
Comment (thibaultcordier) [reply doc/theoretical_description_multilabel_classification.rst]: Be careful some trouble in the documentation (indentation).

**Review by thibaultcordier — CHANGES_REQUESTED** [review]
A few changes are requested to improve the rendering of the documentation.

File: doc/theoretical_description_multilabel_classification.rst (hunk 16 above)
[the remark this answers is not in the document]
Comment (SZiane) [reply doc/theoretical_description_multilabel_classification.rst:171]: Well isn't it better to have all references at the end ?

File: doc/theoretical_description_multilabel_classification.rst (hunk 18 above)
Comment (SZiane) [reply doc/theoretical_description_multilabel_classification.rst:188]: Done

File: doc/theoretical_description_multilabel_classification.rst (hunk 19 above)
Comment (SZiane) [reply doc/theoretical_description_multilabel_classification.rst:191]: Done add space

File: doc/theoretical_description_multilabel_classification.rst (hunk 20 above)
Comment (SZiane) [reply doc/theoretical_description_multilabel_classification.rst:195]: done add space

File: doc/theoretical_description_multilabel_classification.rst (hunk 21 above)
Comment (SZiane) [reply doc/theoretical_description_multilabel_classification.rst:198]: Done add space

File: doc/theoretical_description_multilabel_classification.rst [hunk 22]
```diff
@@ -166,23 +167,36 @@ With :
...
+- First across the collections of functions :math:`(T_\lambda)_{\lambda\in\Lambda}`, we estimate the risk on the calibration data
+  :math:`\{(x_1, y_1), \dots, (x_n, y_n)\}`.
 
-1: First across the collections of functions :math:`(T_\lambda)_{\lambda\in\Lambda}`, we estimate the risk on the calibration data
-\{(x_1, y_1), \dots, (x_n, y_n)\}`.
-2: For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
-:math:`\mathbb{H}_j: R(\lambda_j)>\alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
+- For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
+  :math:`\mathcal{H}_j: R(\lambda_j) > \alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
 is controlled.
```
Comment (thibaultcordier) [inline doc/theoretical_description_multilabel_classification.rst:193]:
```suggestion
- For each :math:`\lambda_j` in a discrete set :math:`\Lambda = \{\lambda_1, \lambda_2,\dots, \lambda_n\}`, we associate the null hypothesis
  :math:`\mathcal{H}_j: R(\lambda_j) > \alpha`, as rejecting the hypothesis corresponds to selecting :math:`\lambda_j` as a point where risk the risk 
  is controlled.
```

**Review by thibaultcordier — APPROVED** [review]
Thank you for your last changes. For me, your PR is ready. Ask @LacombeLouis for his last comments.

File: notebooks/classification/tutorial_multilabel_classification_precision.ipynb (hunk 15 above)
[the remark this answers is not in the document]
Comment (LacombeLouis) [reply notebooks/classification/tutorial_multilabel_classification_precision.ipynb:1]: So for me this is the last thing we need to change.

**Review by LacombeLouis — APPROVED** [review]

**Pull request merged**